util/src/snowflake.rs

```diff
@@ -85,91 +89,91 @@ pub trait Snowflake {
     }
 }
 
-impl Snowflake for ApplicationId {
+impl Snowflake for Id<ApplicationMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
 }
 
-impl Snowflake for AttachmentId {
+impl Snowflake for Id<AttachmentMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
 }
 
-impl Snowflake for AuditLogEntryId {
+impl Snowflake for Id<AuditLogEntryMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
 }
 
-impl Snowflake for ChannelId {
+impl Snowflake for Id<ChannelMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
 }
 
-impl Snowflake for CommandId {
+impl Snowflake for Id<CommandMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
 }
 
-impl Snowflake for EmojiId {
+impl Snowflake for Id<EmojiMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
 }
 
-impl Snowflake for GenericId {
+impl Snowflake for Id<GenericMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
 }
 
-impl Snowflake for GuildId {
+impl Snowflake for Id<GuildMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
 }
 
-impl Snowflake for IntegrationId {
+impl Snowflake for Id<IntegrationMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
 }
 
-impl Snowflake for InteractionId {
+impl Snowflake for Id<InteractionMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
 }
 
-impl Snowflake for MessageId {
+impl Snowflake for Id<MessageMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
 }
 
-impl Snowflake for RoleId {
+impl Snowflake for Id<RoleMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
 }
 
-impl Snowflake for StageId {
+impl Snowflake for Id<StageMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
 }
 
-impl Snowflake for UserId {
+impl Snowflake for Id<UserMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
 }
 
-impl Snowflake for WebhookId {
+impl Snowflake for Id<WebhookMarker> {
     fn id(&self) -> u64 {
         self.get()
     }
```

Comment:
Could these note be collapsed to remove the code duplication?

Reply:
Not all IDs are snowflakes